### ISSUE TITLE
Reformat semicolons in names

### ISSUE THREAD

**Issue by 1ec5** [gravitystorm/openstreetmap-carto#4755]
### Expected behavior

Some features have `name` tags that contain multiple values separated by a semicolon. Some examples from the United States:

* [This `place=town` node](https://www.openstreetmap.org/node/158859890)’s `name` tag contains an English name and a Yiddish name separated by a semicolon. It is also tagged with `name:en` and `name:yi`, but the dual `name` is appropriate because of the widespread use of a language within the town that would be considered a minority language elsewhere.
* [This `amenity=place_of_worship` area](https://www.openstreetmap.org/way/350497025)’s `name` tag contains an Amharic name and an English name separated by a semicolon. Both names are [signposted equally prominently](https://www.mapillary.com/app/?pKey=299319525035558) and used interchangeably. No [`default_language`](https://wiki.openstreetmap.org/wiki/Key:default_language) tag applies in this case, because that key is intended for administrative boundaries, whereas this is a one-off feature.
* [This road](https://www.openstreetmap.org/way/679125385)’s `name` tag contains two English names separated by a semicolon. The road is maintained jointly by two highway departments that disagree on the name, going as far as to post competing street name signs up and down the road. As a result, local residents also disagree on the name.

Unlike in [some countries](https://wiki.openstreetmap.org/wiki/Multilingual_names#United_States_of_America), there hasn’t historically been a consensus to separate dual names with an ad hoc delimiter such as a hyphen or slash. Instead, it’s not uncommon for mappers to use a standard [semicolon value separator](https://wiki.openstreetmap.org/wiki/Semi-colon_value_separator) as they would in any other tag. Apart from consistency with other tags, a semicolon is much less likely to occur within a name in reality.

A mapper who uses the semicolon delimiter would expect a renderer to reformat the semicolon in some fashion. A language-agnostic treatment would be to replace each semicolon with a newline, just as with `ref`s in #750. A newline would be unambiguous because it isn’t possible for a raw tag value to contain a newline.

### Actual behavior

Unfortunately, openstreetmap-carto renders the raw `name` tag verbatim, including the semicolon:

<img src="https://user-images.githubusercontent.com/1231218/208253823-c2e421c8-5434-4a21-84c2-b9f84001b1a4.png" width="400" alt="Kaser"> <img src="https://user-images.githubusercontent.com/1231218/208255424-ad004fee-3684-4f80-9f3d-f2a24563f7d9.png" width="400" alt="Debre Yibabe Kulbi Kidus Gabriel Ethiopian Orthodox Tewahedo Church"> <img src="https://user-images.githubusercontent.com/1231218/208255300-165f4491-d5bb-4a91-bbc4-c9074d3bab5e.png" width="400" alt="Cincinnati Columbus Cincinnati">

Without support for a semicolon delimiter, openstreetmap-carto encourages mappers to choose unpredictable delimiters instead. A previous version of the Kaser node used a slash, indistinguishable from an individual place name or POI name that contains a slash in reality.

### Implementation notes

#750 splits `ref` on `;` and recombines it with `\n`, primarily to choose a shield image based on the length of the longest name. However, a simple `replace()` could suffice for `name` on a point-placed label such as a place or POI.

https://github.com/gravitystorm/openstreetmap-carto/blob/62e8d547a39f8b40e774e98d295b4f9758bc41be/project.mml#L1775-L1781

There’s also a very rare [`;;` escape sequence](https://wiki.openstreetmap.org/wiki/Semi-colon_value_separator#Escaping_with_%27;;%27) for cases where a single name legitimately contains a semicolon. To handle this case, the `replace()` call can be nested inside another `replace()` call that replaces `\n\n` with `;`, or `regexp_replace()` can be called instead.

A newline may not be suitable within line-placed labels (roads, rivers, etc.). In these cases, perhaps an em dash could be used. Though slightly less language-agnostic than a newline, it’s still independent of the writing direction and no less ambiguous to the viewer than a hyphen or slash that’s hardcoded in the database.

/ref #1086

**Comment by imagico** [comment]
First of all i am glad there is a renewed discussion about how OSM should treat multilingual names.  Back in 2017/2018 when several of the maintainers of this style tried to suggest changes in mapping practice to address some of the fundamental difficulties in proper labeling of names (see [here](https://wiki.openstreetmap.org/wiki/Proposed_features/Default_Language_Format), [here](https://wiki.openstreetmap.org/wiki/Proposed_features/Language_information_for_name) and [here](https://imagico.de/blog/en/you-name-it-on-representing-geographic-diversity-in-names/)) the overall consensus among mappers seemed to be that people were fairly content with the status quo.

The relevant issue regarding multilingual name rendering in general by the way is #4404 - where i already commented on (including contemplating the idea to explicitly remove support for the free form compound label painting in the name tag - like with ` / ` or ` - `).

This issue is about the specific suggestion to interpret name tags containing a semicolon separate list of names.  This would depend on how widespread this tagging is in the database.  I have never seen this being used *in the wild* so far.  If anyone could run through a planet file to see how many cases of this we have so far that would be helpful.  For comparison:  Having anything other than a single name in a name tag is exotic in general but the free form compound labels mentioned are moderately widespread in some cases:

* low admin_level administrative boundary relations
* high order populated places in some countries
* large `natural=bay` and `natural=strait` polygons (from the time we incentivized that mapping - see #4546) 
* some POIs which typically don't have a proper name and where mappers instead tag something like 'type + ref' - see also #4629.

To be clear: Since we never explicitly supported that kind of *compound labeling string in name tag* mapping, support for the semicolon separated list would IMO not compete with this to be supported by OSM-Carto.  It would still need to have sufficient (and sufficiently widespread) use to be considered a tagging method that has consensus support by the global mapper community.  Or to say it with different words:  Back in 2014 @matkoniecz on #1086 concluded that storing anything other than a single name in the name tag is not correct tagging.  This might have changed since then - but it still would require evidence that it actually has IMO.  Also consensus support for the semicolon separated list would IMO mandate explicitly removing support for the free form compound labeling strings as discussed in #4404.

Independent of the question if the semicolon separated lists in the name tag have wide support from the world wide mapper community - there is also the other big problem of multilingual names that unfortunately would not be solved by adopting semicolon separated lists in the name tag, that is the *Han unification problem* - see #2208.  Contrary to the name this is not specific to CJK but also occurs with Arabic and Cyrillic scripts - see the issues referencing #2208.

This is largely why - as mentioned in the beginning - several maintainers of this style suggested different approaches to the problem of multilingual names that would address both the problem of specifying more than one name as the locally used name (and potentially their order) and to provide information on what languages these names are actually in to allow using the correct typefaces to render them (without resorting to error prone double tagging and name matching heuristics across multiple tags)

As said - this matter is independent of the specific suggestion of this issue but i would find it unfortunate if mappers form an opinion on if and how to store multiple names in the name tag without being aware of and having had a chance to consider this other big problem of multilingual names.  I would even go as far as saying that the Han unification problem is the larger problem of the two because it affects a much larger number of potential map users (the number of people living in countries where CJK/Arabic/Cyrillic are used is probably much larger than the number of people living in multilingual areas)

**Comment by ZeLonewolf** [comment]
Regarding the specific question of prevalance of semi-colon separations, I ran the following overpass query today:

```
[out:json][timeout:36000];

nwr[name~";"];
out count;
```

This returned the following output:

```
{
  "version": 0.6,
  "generator": "Overpass API 0.7.59 e21c39fe",
  "osm3s": {
    "timestamp_osm_base": "2022-12-17T14:03:07Z",
    "copyright": "The data included in this document is from www.openstreetmap.org. The data is made available under ODbL."
  },
  "elements": [

{
  "type": "count",
  "id": 0,
  "tags": {
    "nodes": "8872",
    "ways": "22077",
    "relations": "1138",
    "total": "32087"
  }
}

  ]
}
```

I have not done any further analysis to characterize precisely how those 32,000 objects are distributed in the database.

**Comment by matkoniecz** [comment]
> Back in 2014 @matkoniecz on https://github.com/gravitystorm/openstreetmap-carto/issues/1086 concluded that storing anything other than a single name in the name tag is not correct tagging. This might have changed since then - but it still would require evidence that it actually has IMO.

it is also possible that I was wrong in 2014. At the very least, it is much more complex in areas where multiple languages are in active use, sometimes without ability to single  language dominating over others.

(ad to that political complexity of declaring one language dominating over other)

**Comment by ZeLonewolf** [comment]
Link to additional discussions in the community forums on this topic can be found here:
https://community.openstreetmap.org/t/multiple-delimited-names-in-the-name-tag

**Comment by imagico** [comment]
32k occurrences of `;` in name tags is a starting point but i think a somewhat deeper look is in order.  Quite a few tools used for imports and automated edit automatically conflate tags with semicolons so looking over these a bit more in detail (in terms of spatial distribution, feature types and origin in particular) would be good.

Based on a few quick looks around central Europe (including some actual multilingual regions) - my impression is that the three main cases of *semicolon in name tag* in those areas are:

* power lines where the name tag seems to be used in a similar fashion as `ref` on roads, i.e. the same physical line is part of several virtual lines/routes and is named based on the names of those routes - using semicolons as necessary.  I am not sure is this is consensus tagging.
* office sharing where the operator/company name is tagged on the name tag - like doctors' offices shared by several doctors or business offices shared by several companies.  I am not sure is this is consensus tagging either.
* a few cases of roads - with either different names on both sides (for which we have name:right/name:left i think), conflations of roads with different names or cases like @1ec5 mentioned where multiple names compete (though usually this is a case of name and alt_name because typically one name dominates in local use).
* quite a large number of obvious incorrect uses of the name tag that are not names at all - lists or codes that belong in `ref`, address components (like street name and number) etc.

In the US there also seems to be cases of this coming from Tiger imports:

https://www.openstreetmap.org/way/5353554
https://www.openstreetmap.org/way/126300618

At a quick look i could not find any occurrence of multilingual names tagged this way.  Can anyone point to an area where this is common locally?

**Comment by aighes** [comment]
> The relevant issue regarding multilingual name rendering in general by the way is #4404 - where i already commented on (including contemplating the idea to explicitly remove support for the free form compound label painting in the name tag - like with `/` or `-`).

I think there is no need to remove any support, it would be totally fine just to add support for `;` so mappers, who actually wants to support the defined delimiter get an acceptable result displayed. By acceptable I mean a delimiter of those names, which humans commonly expect on maps. 

Of course you wont find many `;` in the name at the moment, as most mappers verify their results with OSMcarto and in the end the beautiful map is a higher motivation than following the rule of which delimiter to use.

**Comment by imagico** [comment]
Because i am not sure if i have made that clear enough:  The question that we need to discuss here is not primarily if the semicolon separated list is a more suitable  form to record multiple names in a single name tag than the various free form compound labeling strings entered into name tags of certain features (see https://github.com/gravitystorm/openstreetmap-carto/issues/4755#issuecomment-1356405435).  That is (a) a discussion for a different venue and (b) something that is not really relevant here at all because we never decided to support the free form compound labeling strings and would probably never have done so if we have had the opportunity to choose.

What i would like to know if there is adoption of the *semicolon separated list of names* idea in tagging multiple names in cases where there is consensus among mappers that having multiple names is a suitable use of the name tag and what these names then actually mean.  I listed a few cases where that might be the case in https://github.com/gravitystorm/openstreetmap-carto/issues/4755#issuecomment-1356462854 but also mentioned that it is not clear if these represent consensus tagging.

I in particular so far have not seen any evidence that there is consensus among mappers in multilingual regions, especially those subject to the Han unification problem, that the semicolon separated list is a suitable, let along the desirable form to record multilingual names.  I consider this a relevant question, because - as explained above - we know that there are more elegant (in the sense of less error prone and more flexible) ways to handle these cases.  If mappers in regions where this is an important matter decide to go with the semicolon separated lists despite the known disadvantages i would consider it our obligation to support that.  But so far i see the discussion dominated by people exclusively using latin script and predominantly not from multilingual regions and i am hesitant to consider their view as representative on this matter.

The other, more formal technical question is if a semicolon separated list in the name tag is an ordered or an unordered list.  If it is ordered it would compete with the `name`/`alt_name` tagging scheme - or in other words: we should consider rendering `name=foo1` + `alt_name=foo2` (2M uses) in the same way as this issue suggests rendering `name=foo1;foo2`.  If it is unordered and the semicolon separated list in the name tag is only to be used when the different names have exactly the same weight as the locally used name of the feature then maybe we should consider using a different order than the one tagged (like sorting by script type and secondarily length of name).

In any case - if we should decide to support this it would be essential that QA tools support checking if this tagging is applied in a consistent manner.  That means - for multilingual names - if all the components of the semicolon separated list in the name tag are also found in individual name:<lang> tags.  Because only if there are practically usable QA tools that support checking this can mappers successfully implement and maintain such a tagging scheme in their area.  Does anyone know if any of the commonly used QA tools and the verifiers of editors include this kind of consistency check on name tagging?

To summarize my current understanding of the use/potential use of semicolon separated lists in the name tag, there seem to be the following subtypes of this:

* cases where multiple real world concepts with different names are conflated into one OSM feature - like power lines and shared offices.
* roads with different names on both sides - here the tagging competes with the more commonly used `name:right`/`name:left` (28k uses), which we could think about supporting.
* features with two names from the same language in use locally without there being a verifiable difference in prevalence in use.  This is probably extremely rare.
* features with two names from the same language in use locally but with a clear difference in prevalence.  Here the tagging competes with the more commonly used `alt_name` (2M uses) - which we could think about supporting.
* features in multilingual regions with several languages  in use without there being a verifiable difference in prevalence in use.  I suppose in those cases consensus among those who support the semicolon separated lists is that the individual names are in addition to be tagged in `name:<lang>`.

TL;DR:  Main questions that i would appreciate input on are:

* Are there any multilingual regions where the semicolon separated list in name tag has some adoption in use (or at least is discussed in the local multilingual community as a solution)?
* Are there any QA tools/editor verifiers that check correct use of semicolon separated list in name tag for multilingual names?

**Comment by aighes** [comment]
In multi-lingual areas to me it seems to be the consensus, that `name` should contain a list of values. Consensus listed on https://wiki.openstreetmap.org/wiki/Multilingual_names for Eg. Belgium, Germany, Hong Kong, Italy, Macau, Morocco, New Zealand, Portugal, Slovenia,  Spain, Switzerland (seems only for country name).
But as mentioned above, `;` is not used as a delimiter, because how `name` is handled by the map makers. `;` is a technical delimiter, but not something you expect to have shown on the map. So the mappers have chosen the simple way of just writing the "beautiful" delimiter of their choice in `name`

Based on my knowledge regarding Germany areas, multiple values in `name` are not on a same level as `alt_name`. The main reason having two names is, that they are (eg. for political reason) on same level of importance.

**Comment by adamfranco** [comment]
In the years since this issue was opened, @1ec5 has demonstrated [how vector maps can be elegantly rendered](https://community.openstreetmap.org/t/osm-americana-your-local-language-companion/141757) in a user’s preferred language and developed a MapLibre plugin called “[Diplomat](https://github.com/osm-americana/diplomat/)” to achieve this localization in other map renderings. As improved localization permeates the OSM ecosystem I hope that speakers of minority languages can more fully access OSM in their native tongues.

As has been discussed previously in many forum posts, alternate separators like space (` `), pipe (`|`), or slash (`/`) do appear in single (non-multi-valued) real-world names and cannot be reliably parsed as multi-value separators. See [Name lists in OSM Americana; or: how I learned to love semicolons](https://community.openstreetmap.org/t/name-lists-in-osm-americana-or-how-i-learned-to-love-semicolons/7686) for more background on separators. 

The threads listed above include numerous examples of multi-valued `name=*` tags where there are multiple legitimate primary names for a town, stream, or feature. In many cases these multiple primary names are due to there being multiple local languages, neither of which has more primacy than the other. 

The big impediment to broader usage of the [standard semicolon (;) separator](https://wiki.openstreetmap.org/wiki/Semi-colon_value_separator) between multiple values in the `name=*` tag is lack of resolution of this issue. OSM Carto is currently encouraging contributors to use improper tagging as a work-around to the unsightly rendering of raw name values when `;` is used as a separator.

**Comment by ilias52730** [comment]
I strongly support fixing this.

As an example, in Morocco, French, Tamazight, and Arabic are prominently displayed and equally used on many street signs and institutions across the whole country. This results in incredibly long, multi-script names. We can't give importance to one language or the other, as that would mean giving a secondary role to one of the national cultural identities. Every single street name, admin name, and POI name has 2 or 3 languages in the `name` tag.

Right now, because Carto doesn't handle semicolons, we are forced to mash everything together with spaces like this:
`name=Tanger-Tétouan-Al Hoceïma ⵟⴰⵏⵊ-ⵟⵉⵜⴰⵡⵉⵏ-ⵍⵃⵓⵙⵉⵎⴰ طنجة تطوان الحسيمة`

Or worse:
`name=Université Mohammed VI des Sciences de la Santé ⵜⴰⵙⴷⴰⵡⵉⵜ ⵎⵓⵃⴰⵎⵎⴷ ⵡⵉⵙⵙ ⵚⴹⵉⵚ ⵏ ⵜⵓⵙⵏⵉⵡⵉⵏ ⴷ ⵜⴷⵓⵙⵉ جامعة محمد السادس للعلوم والصحة` (Yes, it's a university. Yes, the 3 languages are prominently displayed, see image below).

<img width="600" alt="Image" src="https://github.com/user-attachments/assets/ffe6bf06-c8d1-4fc7-849e-0e181bc7a572" />

Because Carto only reads the name tag as one single name and slaps it on the renderer, this discourages contributors from using the correct semi-colon tagging, because the map would look like a messy csv file, and causes data on all data consumers and renderers to look like this (screenshot from Americana).

<img width="600" alt="Image" src="https://github.com/user-attachments/assets/b538f8c6-21e6-4e51-b9c3-e7b99ffae63e" />

As a test, I tried putting semi-colons in the name tag for a town in Morocco (Had Soualem) to see how it would look on Americana. See the result for yourself. The language of my browser is shown prominently, while still showing the other languages used, and without uselessly repeating _Had Soualem_.

<img width="600" alt="Image" src="https://github.com/user-attachments/assets/37d69fba-c223-4f32-9ad7-7b48cb495b0d" />

If Carto would also simply support semi-colons, this would make data infinitely better, not only on Carto, but for literally all data consumers and renderers using OSM.

**Comment by imagico** [comment]
This is not the place to promote ideas how you think mappers should tag things and this is especially not the place to try to push OSM-Carto to participate in an attempt to promote such ideas.

I have commented above in 2022 when this issue was opened on OSM-Carto's perspective on the matter.  No substantial discussion took place based on that.  I have now put up an RFC for our strategy in name labeling in OSM-Carto in #5216.  That is - likewise - open for discussion - but not for lobbying for and promoting the specific interests of projects that do not share our goals.

**Comment by zekefarwell** [comment]
> Because Carto only reads the name tag as one single name and slaps it on the renderer, this discourages contributors from using the correct semi-colon tagging, because the map would look like a messy csv file

@ilias52730, It's certainly frustrating, but it is worth dealing with the messy looking labels in the short term if you want support for semi-colon separated names to improve.  Data consumers will only bother to support it if they determine the practice is widespread enough to be worth their time.  It looks like sticking with non-standard delimiters will ultimately end in disappointment anyway if OSM carto stops displaying labels for all multi-value `name` tags as recommended in https://github.com/openstreetmap-carto/openstreetmap-carto/issues/5216:

> - OSM-Carto should stop displaying labels (and thereby providing supportive feedback) directly from a name tag if that tag can be determined not to contain a single proper name
> ...
> [Indicated by:]
> - Combination of unicode ranges for different scripts
> - Presence of character sequences that are used as separators between different components (primarily `/` , `-`, `<newline>` and `;`)

**Comment by imagico** [comment]
By the way - analyzing use of name tags i noticed that the most common value of Latin script name tags with a semicolon is not a compound name consisting of several individual names for the feature in question:

https://taginfo.openstreetmap.org/tags/name=California%20High%20Speed%20Rail%3BDesert%20Subdivision

This is not really an argument either for or against interpreting the semicolon as a delimiter in name tags IMO, but it is important to realize i think that the semicolon is practically not always used in name tags in such role.

**Comment by 1ec5** [comment]
> https://taginfo.openstreetmap.org/tags/name=California%20High%20Speed%20Rail%3BDesert%20Subdivision

A frequency analysis will turn up mostly navigable ways because they get split so often. The most frequent values don’t necessarily reflect usage trends in terms of distinct features. I [fixed](https://www.openstreetmap.org/changeset/182096907) this proposed railway by moving the first name into `operator=*` where it belongs.[^operator] It’s also correctly on the route relations. The offending changeset introduced lots of other naming problems that remain, but fortunately proposed rails won’t render in OSM Carto. There isn’t even consensus that these rails should be mapped yet.

> In the US there also seems to be cases of this coming from Tiger imports:

This didn’t come from TIGER. Potlatch&nbsp;1 used to make it far too easy to [accidentally merge](https://www.openstreetmap.org/way/25304581/history/2) two roadways, concatenating their `name=*` and other tags with `; ` (a spaced semicolon). It should be on the decline as mappers gradually discover and [clean up](https://www.openstreetmap.org/way/126300618/history/11) these blatant errors.[^tiger] This phenomenon was almost exclusive to roads and paths. Other feature types like places, shops, or buildings were spared because Potlatch didn’t consider them mergeable. I don’t think it affected power lines because we weren’t generally naming them back then.

[^operator]: In reality, North American railways are named by subdivision, so that’s how they’re tagged. But the traveling public only knows them by operating companies, so that’s what transportation maps are expected to show. Occasionally a newcomer asks the local community why OSM Carto labels railways by `name=*` instead of what’s in `operator=*` or `operator:short=*`, which would be consistent with conventional maps, and we have to explain that it’s a global style. I’m surprised this discussion hasn’t played out here in this repository before.
[^tiger]: Among roads in the U.S. that were originally imported from TIGER, the number of affected ways has come down to [under 100](https://qlever.dev/osm-planet/bFvZrg). The actual number of roads is somewhat higher, probably in the [low hundreds](https://qlever.dev/osm-planet/IjRvva), considering way splitting but also iD’s tendency to strip out the space.

**Comment by imagico** [comment]
> A frequency analysis will turn up mostly navigable ways because they get split so often. The most frequent values don’t necessarily reflect usage trends in terms of distinct features.

That is indeed something very important to keep in mind when analyzing name data - divisible elements with name tags (roads, railway lines, border lines etc.) will often have multiple occurrences of the same name while placenames and POIs will typically only have one.  This creates a bias towards the divisible features in any overall name analysis.

Of the 40M distinct `name` values in the database about 10M have more than one occurrence.  That, of course, includes real world duplicate names, but also split roads etc. in substantial numbers.

Current number of `name` tags with semicolon is slightly above 36k (compared to 32k in 2022).  The total number of compound name tags without delimiter or with other delimiters is difficult to precisely assess without a more complex analysis, but it is likely to be somewhere between 500k and 2M.  All that relative to a total number of `name` tags of 113M.

Looking at name tags with a semicolon just among node features will give a less biased impression - though selectively only for node features of course - search for `;` in

https://taginfo.openstreetmap.org/keys/name?filter=nodes#values

and you will get some idea of the spectrum of use cases of the semicolon in `name` tags on nodes.

There is another use case i have not seen much with other separators and that i had not mentioned above in https://github.com/openstreetmap-carto/openstreetmap-carto/issues/4755#issuecomment-1360248036.  That is use of the `name` as a search term list and essentially listing different versions of the same name.  For example:

https://www.openstreetmap.org/node/8764643819
https://www.openstreetmap.org/node/4262658989
https://www.openstreetmap.org/node/2380050038

I have not seen that in a similar frequency with the other delimiters, this seems fairly specific to the semicolon.
